presto-iceberg/src/test/java/com/facebook/presto/iceberg/nessie/TestNessieMultiBranching.java

```diff
@@ -38,6 +40,7 @@
 import static com.facebook.presto.tests.QueryAssertions.assertEqualsIgnoreOrder;
 import static org.assertj.core.api.Assertions.assertThat;
 
+@Test(singleThreaded = true)
```

Comment:
I believe you have made it single-threaded execution because   `resetData` will clean all the branches including the concurrently executing test case's branches.

So, maybe the efficient way is to have testcase level clean-up which only drop the branches it created (instead of generic cleanup) and ensuring none of the testcase commits on default branch.

Reply:
I'm not a fan of doing cleanup within a test method itself, because you'd have to wrap the entire test method code in a `try-finally` to guarantee that cleanup runs at the end of the method. 
This test class has 3 tests in total, so running this single-threaded just makes the test more predictable and less prone to concurrent cleanup/execution and shouldn't affect CI time too much